### PR TITLE
Pull build secrets from AWS

### PR DESCRIPTION
**Pull request by strutt**
----

## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: azure/templates/identity-service-templating.yml [hunk 1]
```diff
@@ -3,16 +3,59 @@ steps:
...
+    displayName: "Get AWS config"
+
   - bash: |
       set -euo pipefail
+      
+      for secret_name in APIGEE_OTP_KEY \
+                         APIGEE_PASSWORD \
+                         ; do
+          secret_id="ptl/azure-devops/apigee-nonprod/${secret_name}"
+          secret_details=$(aws --profile apm_ptl secretsmanager get-secret-value --secret-id ${secret_id})
```
Comment (vurt007) [inline azure/templates/identity-service-templating.yml:30]: no need to have two steps here .. use

aws --profile apm_ptl secretsmanager get-secret-value --secret-id ${secret_id} --query SecretString --output text

File: azure/templates/identity-service-templating.yml [hunk 2]
```diff
@@ -3,16 +3,59 @@ steps:
...
+      chmod 400 ~/.aws/config
+    displayName: "Get AWS config"
+
   - bash: |
       set -euo pipefail
+      
+      for secret_name in APIGEE_OTP_KEY \
+                         APIGEE_PASSWORD \
+                         ; do
+          secret_id="ptl/azure-devops/apigee-nonprod/${secret_name}"
```
Comment (vurt007) [inline azure/templates/identity-service-templating.yml:29]: how does this support prod deploys ?

File: azure/templates/identity-service-templating.yml [hunk 3]
```diff
@@ -3,16 +3,59 @@ steps:
...
+  - bash: |
+      set -euo pipefail
+      
       export SERVICE_ARTIFACT_NAME=`ls $(Pipeline.Workspace)/$(SERVICE_NAME)`
       echo "##vso[task.setvariable variable=SERVICE_ARTIFACT_NAME]$SERVICE_ARTIFACT_NAME"
-    displayName: 'Set SERVICE_ARTIFACT_NAME'
+
+      # AWS cli can handle getting up to 10 params in a single call.
+      params=$(aws --profile apm_ptl ssm get-parameters --names \
+                /ptl/azure-devops/apigee-nonprod/APIGEE_USERNAME \
```
Comment (vurt007) [inline azure/templates/identity-service-templating.yml:45]: as above .. how do this deploy to prod ?

File: azure/templates/identity-service-templating.yml [hunk 4]
```diff
@@ -3,16 +3,12 @@ steps:
     displayName: 'Use Python 3.8'
     inputs:
       versionSpec: 3.8
-
+
```
Comment (Sourceless) [inline azure/templates/identity-service-templating.yml:6]: Please remove random whitespace

File: azure/azure-release-pipeline.yml [hunk 5]
```diff
@@ -187,8 +233,14 @@ stages:
   dependsOn: 
     - DeployRef
   variables:
```
Comment (Sourceless) [inline azure/azure-release-pipeline.yml:235]: The SERVICE_BASE_PATH for this environment needs to be different -- `oauth2-no-smartcard`

**Pull request merged**